### PR TITLE
[NEAT-310]🏰Large Example

### PR DESCRIPTION
**Pull request by doctrino**

### CODE REVIEW

File: cognite/neat/rules/importers/_inference2rules.py [hunk 1]
```diff
@@ -36,28 +36,36 @@ class InferenceImporter(BaseImporter):
     """Infers rules from a triple store.
 
     Rules inference through analysis of knowledge graph provided in various formats.
-    Use the factory methods to create an triples store from sources such as
+    Use the factory methods to create n triple store from sources such as
```
Comment (doctrino) [inline cognite/neat/rules/importers/_inference2rules.py:39]:
```suggestion
    Use the factory methods to create a triple store from sources such as
```

File: cognite/neat/graph/extractors/_classic_cdf/_assets.py [hunk 2]
```diff
@@ -21,17 +22,28 @@ class AssetsExtractor(BaseExtractor):
...
         to_type (Callable[[Asset], str | None], optional): A function to convert an asset to a type. Defaults to None.
             If None or if the function returns None, the asset will be set to the default type "Asset".
+        total (int, optional): The total number of assets to load. If passed, you will get a progress bar if rich
+            is installed. Defaults to None.
+        limit (int, optional): The maximal number of assets to load. Defaults to None. This is typically used for
+            testing setup of the extractor. For example, if you are extracting 100 000 assets, you might want to
+            limit the extraction to 1000 assets to test the setup.
     """
 
+    _SPACE_PATTERN = re.compile(r"\s+")
```
Comment (nikokaoja) [inline cognite/neat/graph/extractors/_classic_cdf/_assets.py:32]: As values might contain non-alpha num characters that can ill form URI, suggestion to remove any character that will not fit end View/Container id:

r"(^[a-zA-Z][a-zA-Z0-9_]{0,253}[a-zA-Z0-9]?$)"

We have in :
https://github.com/cognitedata/neat/blob/main/cognite/neat/rules/models/_types/_base.py#L4C1-L4C25

Various compliance regexes (which I guess should be move to utils)

File: cognite/neat/graph/extractors/_classic_cdf/_assets.py [hunk 3]
```diff
@@ -50,21 +71,45 @@ def from_hierarchy(
...
+                from rich.progress import track
+            except ModuleNotFoundError:
+                to_iterate = self.assets
+            else:
+                to_iterate = track(self.assets, total=self.limit or self.total, description="Extracting Assets")
+        else:
+            to_iterate = self.assets
+        for no, asset in enumerate(to_iterate):
             yield from self._asset2triples(asset, self.namespace)
+            if self.limit and no >= self.limit:
```
Comment (nikokaoja) [inline cognite/neat/graph/extractors/_classic_cdf/_assets.py:111]: this is super useful for testing

File: cognite/neat/graph/extractors/_classic_cdf/_assets.py [hunk 4]
```diff
@@ -74,6 +119,8 @@ def _asset2triples(self, asset: Asset, namespace: Namespace) -> list[Triple]:
         type_ = "Asset"
         if self.to_type:
             type_ = self.to_type(asset) or type_
+        type_ = self._SPACE_PATTERN.sub("_", type_)
```
Comment (nikokaoja) [inline cognite/neat/graph/extractors/_classic_cdf/_assets.py:122]: see my comment about potential ill-formed URIs due to illegal characters beyond space.
Option could be to form URI safe , but ideally remove/replace any none websafe/dms-safe character with e.g. "_"

File: cognite/neat/rules/importers/_inference2rules.py [hunk 5]
```diff
@@ -262,12 +275,9 @@ def _update_value_type_occurrence_in_comment(cls, value_type: str, comment: str)
 
     @classmethod
     def _read_value_type_occurrence_from_comment(cls, value_type: str, comment: str) -> int:
```
Comment (nikokaoja) [inline cognite/neat/rules/importers/_inference2rules.py:277]: Should be a task in our board to track occurence not as in comment but some private variable as regex is expensive

**Review by nikokaoja — APPROVED** [review]

**Pull request merged**